.github/workflows/tests.yml

```
@@ -3,23 +3,63 @@ name: tests
 on:
   push:
   pull_request:
+    branches:
+      - main
+  workflow_dispatch:
   schedule:
-      - cron: '00 4 * * *'  # daily at 4AM
+      - cron: '0 0 * * 0'  # weekly
 
 jobs:
-  build:
+  unit-tests:
 
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.9"] # TODO: expand versions
 
       fail-fast: false
     steps:
 
     - uses: actions/checkout@v4
 
+    # TODO (maffettone): Change host after CD into bluesky/bluesky-pods
+    - name: Download and build bluesky-pods
+      run: |
+        docker pull ghcr.io/maffettone/bluesky-pods-bluesky:latest
+        docker tag ghcr.io/maffettone/bluesky-pods-bluesky:latest bluesky:latest
+
+    - name: Start Bluesky containers
+      run: |
+        pushd bluesky_adaptive/tests/podman
+        docker-compose up -d
+        popd
+
+    - name: Log for are containers running?
+      run: docker ps -a
+
+    - name: Check container status and fetch logs if any failed
+      run: |
+        failed_containers=()
+        while read -r container; do
+          status=$(docker inspect --format='{{.State.Status}}' "$container")
+          if [ "$status" != "running" ]; then
+            echo "::error ::Container $container has status $status"
+            failed_containers+=("$container")
+          fi
+        done < <(docker ps -aq)
```

Typo maybe?  I'm not familiar with `< <(...`

```suggestion
        done < $(docker ps -aq)
```

---

I needed the same explanation. 

> Process substitution (< <(command)) allows for the output of a command to be treated as a file. The syntax < <(command) effectively directs the output of command into the while loop as if it were being read from a file. This is particularly useful for reading the output line by line in a loop.
> 
> Command substitution ($(command)) captures the output of a command and can be used as an argument to another command. However, using done < $(command) is actually a misuse of syntax and would result in an error because < expects a file name to follow it, not a string of output. If you were to attempt to use command substitution to redirect input to a loop, you might mistakenly concatenate all output into a single input, which is not the intention here.

---

Today I learned about `<()`.  Thanks!